### PR TITLE
Simplify the chain context

### PR DESCRIPTION
**Pull request by wmjrosenblatt**
## Problem

The front end chain context is very complex right now and utilizes 3 versions of the chainspec to keep track of changes.

## Solution
Slightly change how the context manages the state to reduce it down to 2 versions of the chainspec

### CODE REVIEW

**Comment by wmjrosenblatt** [comment]
Seems to work but please someone else verify locally that everything is working okay.

File: app/src/contexts/ChainSpecContext.tsx [hunk 1]
```diff
@@ -58,57 +56,48 @@ export const ChainSpecProvider: React.FC<ChainSpecProviderProps> = ({ children }
...
-
   // Dirty spec receives all updates to the spec from the UI (ie. onchange events)
   const [dirtyChainSpec, setDirtyChainSpec] = useState<ChainSpec | null>(null);
 
   const [nextChainId, setNextChainId] = useState<number>(0);
   const [revision, setRevision] = useState<string|null>(null);
   const [isInteracting, setIsInteracting] = useState<boolean>(false);
 
-  const insertChainSpec = useCallback((type: string, chainId: number, index: number): void => {
+  const insertChainSpec = (type: string, chainId: number, index: number): void => {
```
Comment (acruikshank-wm) [inline app/src/contexts/ChainSpecContext.tsx:66]: BLOCKING: I don't think this is a good idea. The best case is that we end up regenerating this function and re-rendering other components too often. The worst case is we end up with a component left with a callback with a stale reference to the dirty chain or chain id. Is there a specific reason why you are removing these `useCallback`s?

File: app/src/contexts/ChainSpecContext.tsx [hunk 2]
```diff
@@ -58,57 +56,48 @@ export const ChainSpecProvider: React.FC<ChainSpecProviderProps> = ({ children }
...
+  const findChainInCurrentSpec = (chainId: number): ChainSpec | undefined => {
     if (!dirtyChainSpec) {
       return undefined;
     }
     return findByChainId(chainId, dirtyChainSpec);
-  }, [dirtyChainSpec]);
+  };
 
-  const deleteChain = useCallback((chainId: number): void => {
+  const deleteChain = (chainId: number): void => {
```
Comment (acruikshank-wm) [inline app/src/contexts/ChainSpecContext.tsx:89]: see above.

File: app/src/contexts/ChainSpecContext.tsx [hunk 3]
```diff
@@ -58,57 +56,48 @@ export const ChainSpecProvider: React.FC<ChainSpecProviderProps> = ({ children }
...
     const newSpec = deleteChainSpec(dirtyChainSpec, chainId);
-    setChains(newSpec);
-  }, [dirtyChainSpec]);
+    setDirtyChain(newSpec);
+  };
 
-  const updateDirtyChainSpec = useCallback((spec: ChainSpec): void => {
+  const updateDirtyChainSpec = (spec: ChainSpec): void => {
```
Comment (acruikshank-wm) [inline app/src/contexts/ChainSpecContext.tsx:94]: see above

**Review by acruikshank-wm — CHANGES_REQUESTED** [review]
I don't think you should be removing the useCallbacks. That change does not seem to specifically address the problem you are solving. In the future, please try to call out and justify these tangential changes in the PR notes.

**Pull request closed without merging**